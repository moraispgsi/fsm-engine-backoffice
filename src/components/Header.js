/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import styled from 'styled-components';

import Navbar from './Navbar/Navbar';
import Logo from './Navbar/Logo';
import UserMenu from './Navbar/UserMenu';
import NavItem from './Navbar/NavItem';

const StyledHeader = styled.header`
  /* clearfix */
  &:before, &:after {
    display: table;
    content: " ";
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }
  &:after {
    clear: both;
  }

  position: ${props => (props.fixed ? 'fixed' : 'relative')};
  width: ${props => (props.boxed ? '1024px' : '100%')};
  max-height: 100px;
  z-index: 1030;

  /* theme */
  ${props => props.theme.headerBoxShadow && `
    -webkit-box-shadow: ${props.theme.headerBoxShadow};
    box-shadow: ${props.theme.headerBoxShadow};
  `}
`;

const Header = props => (
  <StyledHeader fixed={props.fixed} >
    <Logo
      collapse={props.sidebarCollapse}
      sidebarMini={props.sidebarMini}
      onClick={props.logoOnClick}
      href={props.logoHref}
      logoLg={props.logoLg}
      logoSm={props.logoSm}
    />
    <Navbar
      toggle={props.sidebarToggle}
      collapse={props.sidebarCollapse}
      sidebarMini={props.sidebarMini}
    >
      {props.children}
    </Navbar>
  </StyledHeader>
);

Header.propTypes = {
  children: React.PropTypes.node,
  fixed: React.PropTypes.bool,
  logoOnClick: React.PropTypes.func,
  logoHref: React.PropTypes.string,
  logoLg: React.PropTypes.element,
  logoSm: React.PropTypes.element,
  sidebarMini: React.PropTypes.bool,
  sidebarCollapse: React.PropTypes.bool,
  sidebarToggle: React.PropTypes.func.isRequired,
};

Header.defaultProps = {
  fixed: false,
  sidebarMini: false,
  sidebarCollapse: false,
  logoLg: <span><b>Admin</b>SBCS</span>,
  logoSm: <span><b>A</b></span>,
};

Header.UserMenu = UserMenu;
Header.Item = NavItem;

export default Header;
