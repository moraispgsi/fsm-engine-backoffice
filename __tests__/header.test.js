/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import red from '../src/styles/skin-red';
import Header from '../src/components/Header';

describe('<Header />', () => {
  it('renders correctly with default options', () => {
    const component = renderer.create(<Header sidebarToggle={() => null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with theme', () => {
    const component = renderer.create(<ThemeProvider theme={red}>
      <Header sidebarToggle={() => null} />
    </ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with fixed option', () => {
    const component = renderer.create(<Header fixed sidebarToggle={() => null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with collapsed option', () => {
    const component = renderer.create(<Header sidebarCollapse sidebarToggle={() => null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with fixed and collapsed options', () => {
    const component = renderer.create(
      <Header fixed sidebarCollapse sidebarToggle={() => null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders given logo onClick function', () => {
    const func = () => null;
    const wrapper = mount(<Header logoOnClick={func} sidebarToggle={() => null} />);
    expect(wrapper.find('[name="navbar-logo"]').props().onClick).toEqual(func);
  });

  it('renders given logo href', () => {
    const href = '/some/link';
    const wrapper = mount(<Header logoHref={href} sidebarToggle={() => null} />);
    expect(wrapper.find('[name="navbar-logo"]').props().href).toEqual(href);
  });

  it('renders correctly with given large logo', () => {
    const component = renderer.create(
      <Header
        logoLg={<span>Large</span>}
        sidebarToggle={() => null}
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with given small logo', () => {
    const component = renderer.create(
      <Header
        logoSm={<span>Small</span>}
        sidebarToggle={() => null}
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
