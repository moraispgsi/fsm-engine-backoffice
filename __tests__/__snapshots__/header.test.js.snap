exports[`<Header /> renders correctly with collapsed option 1`] = `
<header
  className="Sdzae">
  <a
    className="cykMTm"
    href="/"
    name="navbar-logo">
    <span
      className="ibQovi"
      name="navbar-logo-lg">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      name="navbar-logo-sm">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      name="sidebar-toggle"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe"
        name="navbar-menu-wrapper" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with default options 1`] = `
<header
  className="Sdzae">
  <a
    className="cykMTm"
    href="/"
    name="navbar-logo">
    <span
      className="ibQovi"
      name="navbar-logo-lg">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      name="navbar-logo-sm">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      name="sidebar-toggle"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe"
        name="navbar-menu-wrapper" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with fixed and collapsed options 1`] = `
<header
  className="glmcVm">
  <a
    className="cykMTm"
    href="/"
    name="navbar-logo">
    <span
      className="ibQovi"
      name="navbar-logo-lg">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      name="navbar-logo-sm">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      name="sidebar-toggle"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe"
        name="navbar-menu-wrapper" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with fixed option 1`] = `
<header
  className="glmcVm">
  <a
    className="cykMTm"
    href="/"
    name="navbar-logo">
    <span
      className="ibQovi"
      name="navbar-logo-lg">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      name="navbar-logo-sm">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      name="sidebar-toggle"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe"
        name="navbar-menu-wrapper" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with given large logo 1`] = `
<header
  className="Sdzae">
  <a
    className="cykMTm"
    href="/"
    name="navbar-logo">
    <span
      className="ibQovi"
      name="navbar-logo-lg">
      <span>
        Large
      </span>
    </span>
    <span
      className="hboVZB"
      name="navbar-logo-sm">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      name="sidebar-toggle"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe"
        name="navbar-menu-wrapper" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with given small logo 1`] = `
<header
  className="Sdzae">
  <a
    className="cykMTm"
    href="/"
    name="navbar-logo">
    <span
      className="ibQovi"
      name="navbar-logo-lg">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      name="navbar-logo-sm">
      <span>
        Small
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      name="sidebar-toggle"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe"
        name="navbar-menu-wrapper" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with theme 1`] = `
<header
  className="Sdzae">
  <a
    className="dXzCJS"
    href="/"
    name="navbar-logo">
    <span
      className="ibQovi"
      name="navbar-logo-lg">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      name="navbar-logo-sm">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="rOyox">
    <button
      className="fa fa-bars jatJFQ"
      name="sidebar-toggle"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe"
        name="navbar-menu-wrapper" />
    </div>
  </nav>
</header>
`;
